src/app/shared/filter/date/custom-date-range.component.ts

```diff
@@ -25,8 +26,16 @@ export class CustomDateRangeComponent {
   @Input() interval: IntervalModel;
   @Input() beginDate: Date;
   @Input() endDate: Date;
+  @Input() podcast: PodcastModel; // podcast is only here for google events tracking on the timepicker
   @Output() customRangeChange = new EventEmitter<DateRangeModel>();
 
+  constructor(public angulartics2: Angulartics2) {}
+
+  onBeginTimeChange(date: Date) {
+    this.onBeginDateChange(date);
+    this.googleAnalyticsEvent('begin-time', date.getUTCHours());
```

Comment:
so when begin or end time changes, this event will fire and so will the `'custom-date'` from the date range component, right?

Reply:
yes, and I'm ok with that. There will be two events that mean two different things we're trying to track: They used the custom date control, and they used the time picker.